src/main/java/com/treasure/hunt/utils/JTSUtils.java

```diff
@@ -24,6 +24,8 @@
      */
     public static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory(new PrecisionModel(1000000000));
 
+    public static final double DISTANCE_ZERO = 1e-10;
```

Comment:
There already is a method called JTSUtils.doubleEqual(a,b) which can be used to achieve the same thing 
(When checking if x equals zero one can call JTSUtils.doubleEqual(x,0) and the method rounds a on the grid of the precision model of the GEOMETRY_FACTORY to determine wether the value is 0 or not).
You can manipulate the value in the construction of the PrecisionModel to get lower precision (use a lesser value then instead)

Reply:
Thank you for pointing that out, I didn't know that. Of course, I've implemented these changes directly.